src/Components/Input.tsx

```diff
@@ -19,21 +19,18 @@ export interface InputProps extends React.HTMLProps<HTMLInputElement> {
  * The `title` and `description` props are rendered above the input.
  *
  */
-export const Input: SFC<InputProps> = ({
-  error,
-  title,
-  description,
-  ...rest
-}) => {
-  return (
-    <Container>
-      {title && <Title>{title}</Title>}
-      {description && <Description>{description}</Description>}
-      <StyledInput hasError={!!error} {...rest} />
-      {error && <InputError>{error}</InputError>}
-    </Container>
-  )
-}
+export const Input: React.ComponentType<InputProps> = React.forwardRef(
+  ({ error, title, description, ...rest }, ref) => {
+    return (
+      <Container>
+        {title && <Title>{title}</Title>}
+        {description && <Description>{description}</Description>}
+        <StyledInput ref={ref} hasError={!!error} {...rest} />
+        {error && <InputError>{error}</InputError>}
+      </Container>
+    )
+  }
+)
```

Comment:
@mzikherman This is what needed to be done to deal with the ref that react-autowhataver wants you to forward to your custom input. See https://reactjs.org/docs/forwarding-refs.html

Reply:
Gah, I spent so long trying to figure that out! I'm hopeful this also clears up an error in an edge case I was coming across in testing (from the autosuggest lib, clicking on the custom rendered suggestions container while its open, which only happened when using a custom input renderer) where an input ref was undefined. Thank you!!

Reply:
Ah, damn, sorry for not noticing earlier that you were struggling with this… let me know next time!

I made a PR for the test I think you mean https://github.com/artsy/reaction/pull/2036

Reply:
@damassi this wound up fixing that `ref` issue we were looking at when using a custom render for the input.

Reply:
Nice, I haven't seen `React.forwardRef` before. (There's a deluge of new React API's coming down the line I need to get on top of.)